actions/shared-node-cache/action.yml

```diff
@@ -5,26 +5,40 @@ inputs:
     description: 'Node version to use'
     required: false
     default: '20.x'
+  cache-cypress:
+    description: 'If true, caches and restore Cypress packages'
+    required: false
+    default: 'false'
+
 runs:
   using: "composite"
   steps:
-    - name: Use Node.js ${{ inputs.node-version }}
-      uses: actions/setup-node@v4
+    - uses: pnpm/action-setup@v4
       with:
-        node-version: ${{ inputs.node-version }}
+        run_install: false
 
-    - name: Cache node_modules and cypress if it exists
-      uses: actions/cache@v4
-      id: cache-node-modules
+    # MUST be before we install node_modules as that depends on finding
+    # Cypress binaries in this cache!
+    - uses: actions/cache@v4
+      if: inputs.cache-cypress == 'true'
       with:
-        path: |
-          node_modules
-          ~/.cache/Cypress
-        key: ${{ runner.os }}-${{ inputs.node-version }}-v1-${{ hashFiles('**/yarn.lock') }}
+        path: ~/.cache/Cypress
+        # Generate a new cache whenever the lock file changes
+        key: ${{ runner.os }}-cypress-${{ hashFiles('**/pnpm-lock.yaml') }}
+        # If source files changed, rebuild from a prior cache.
         restore-keys: |
-          ${{ runner.os }}-${{ inputs.node-version }}-v1-
+          ${{ runner.os }}-cypress-
+
+    - uses: actions/setup-node@v4
+      with:
+        node-version: ${{ inputs.node-version }}
+        cache: pnpm
```

Comment:
So I note that this doesn't have the ssh agent setup step that we run in webapp:
https://github.com/Khan/webapp/blob/master/.github/actions/shared-node-cache/action.yml

We'll need that step for any private repo that we run this in, where we import private dependencies (like event-schemas, for example). It might be nice to add in an option for passing in the ssh key and if it exists, use it to initialize the ssh agent.

Reply:
Agreed.  I'll add that option!